engines/sci/graphics/view.cpp

```diff
@@ -820,12 +795,12 @@ void GfxView::draw(const Common::Rect &rect, const Common::Rect &clipRect, const
 	const int16 width = MIN(clipRect.width(), celWidth);
 	const int16 height = MIN(clipRect.height(), celHeight);
 
-	bitmap += (clipRect.top - rect.top) * celWidth + (clipRect.left - rect.left);
+	const byte *bitmapData = bitmap.getUnsafeDataAt((clipRect.top - rect.top) * celWidth + (clipRect.left - rect.left), width * height);
```

Comment:
Good catch. Both these equations actually would break for clipped views if height is zero; the solution to me seems to be to just to return early if width or height are zero (since nothing gets drawn in that case anyway), and use `celWidth * (height - 1) + width`, which is more accurate (in a hypothetical edge case where view data is truncated mid-line, after the `width`). So, I’ve made that change.